src/pages/search/Search.tsx

```diff
@@ -29,14 +32,75 @@ const Search = () => {
   );
   return (<Fade>
     <Query query={SEARCH_LISTINGS} variables={{ input, ...LISTING_CARD_IMAGE_DIMENSIONS }}>
-      {({ loading, error, data }) =>
-        loading ? <Loading /> :
-        error ? <h2>Error: {error.message}</h2> :
-        <SearchPage listings={data.searchListings} />
-      }
+      {({ loading, error, data }) => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
+          );
+        }
+
+        if (error) isBadUserInputError(error) ? <SearchErrorPage /> : <ErrorPage error={error} />;
```

Comment:
Looks like this is missing a `return`

`http://localhost:4200/work/search?locationQuery=` gives "We're sorry — something's gone wrong."

When I add a `return` statement I instead see the (really nice-looking!) "Please enter in a city or region to see results" page

Reply:
🙏